### PR TITLE
Issue 117 fix, updated setMetadata to use partialInstanceUpdate.

### PR DESCRIPTION
**Pull request by muraliQlogic**
Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/instance.js [hunk 1]
```diff
@@ -764,9 +764,16 @@ Instance.prototype.getTablesStream = common.paginator.streamify('getTables');
...
  *
  * @param {object} metadata Metadata object.
- * @param {string} metadata.displayName The descriptive name for this
- *     instance as it appears in UIs. It can be changed at any time, but
- *     should be kept globally unique to avoid confusion.
+ * @param {Object} metadata.instance
+ * The Instance which will (partially) replace the current value.
+ *
+ * This object should have the same structure as [Instance]{@link google.bigtable.admin.v2.Instance}
+ * @param {Object} metadata.updateMask
```
Comment (sduskis) [inline src/instance.js]: `updateMask` should be optional.  The default ought to be to update all fields.

Comment (muraliQlogic) [reply src/instance.js]: Updated as suggested. Please review.

File: src/instance.js [hunk 2]
```diff
@@ -956,10 +968,14 @@ Instance.prototype.getTablesStream = common.paginator.streamify('getTables');
  * const bigtable = new Bigtable();
  * const instance = bigtable.instance('my-instance');
  *
+ * var instanceToUpdate = {
+ * displayName: 'updated-name'
```
Comment (kolodny) [inline src/instance.js]: nit, index += 2

File: src/instance.js [hunk 3]
```diff
@@ -976,12 +992,21 @@ Instance.prototype.setMetadata = function(metadata, gaxOptions, callback) {
     callback = gaxOptions;
     gaxOptions = {};
   }
+  var updateMask = metadata.updateMask
```
Comment (kolodny) [inline src/instance.js]: Only set items that exist on the metadata object. Follow the same pattern as https://github.com/googleapis/nodejs-bigtable/blob/c2ce475afa903939419fb708837b4692a7df0b76/src/app-profile.js#L364-L375

File: src/instance.js [hunk 4]
```diff
@@ -941,9 +941,21 @@ Instance.prototype.getTablesStream = common.paginator.streamify('getTables');
...
+ * @param {string} [metadata.displayName]
+ * The descriptive name for this instance as it appears in UIs. It can be changed at any time, but
  *     should be kept globally unique to avoid confusion.
+ * @param {string} [metadata.type] The type of the instance. Options are
+ *     'production' or 'development'.
+ * @param {Object.<string, string>} [metadata.labels]  Labels are a flexible and lightweight mechanism for organizing cloud
+ *     resources into groups that reflect a customer's organizational needs and
+ *     deployment strategies. They can be used to filter resources and aggregate
+ *     metrics.
+ * @param {Object} [metadata.updateMask]
```
Comment (kolodny) [inline src/instance.js]: This should not be exposed at all, see below for my other comment about this

**Comment by muraliQlogic** [comment]
Closing this pull request and created [PR#167](https://github.com/googleapis/nodejs-bigtable/pull/167)  with new branch and changes as per @kolodny review comments..

**Pull request closed without merging**